File: Python/horner.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
#
#  horner.py


def horner(k, tbwsp, x):
    wynik = tbwsp[0]
    for i in range(1, k + 1):
        wynik = wynik * x + tbwsp[i]

    return wynik


def horner_rek(k, tbwsp[], x):
    if k == 0:
        return tbwsp[] * x
    return horner_rek(k - 1, tbwsp, x) * x + tbwsp[k]


def main(args):
    k = 3
    tbwsp = []
    x = int(input("Podaj liczbę "))

    for i in range(0, 4):
        tmp = int(input("Podaj wspolczynnik "))
        tbwsp.append(tmp)
        #tbwsp[4]=int(input("Podaj współczynniki "))

    print("Wynik ", horner(k, tbwsp, x))
    print("Wynik ", horner_rek(k, tbwsp, x))


if __name__ == '__main__':
    import sys
    sys.exit(main(sys.argv))

File: cpp/Horner.cpp
/*
 * Horner.cpp
 */


#include <iostream>

using namespace std;

// W(x) = 2x^3 + 3x^2 + 5x + 4 (8)
// W(x) = x (2x^2 + 3x + 5) + 4
// W(x) = x (x(2x + 3) + 5) + 4 (3 mnożenia)

float horner_rek(int k, float tbwsp[],float x)
{
    if (k == 0)
        return tbwsp[0] * x;
    return horner_rek(k - 1, tbwsp, x) * x + tbwsp[k];
    
    }

float horner_it(int k, float tbwsp[], float x){
    float wynik = tbwsp[0];
    for (int i = 1; i < k + 1; i++){
        wynik = wynik * x + tbwsp[i];
    
    return wynik;
}  



int main(int argc, char **argv)
  
    float x = 0;
    float wspolczynniki[4];
    int stopien = 3;
    cout << "Podaj x: " << endl;
    cin >> x;
    
    for(int i = 0; i < 4 ; i ++){
        cout << "Podaj współczynnik: ";
        cin >> wspolczynniki[i];
    }
        
    cout << "Wartość wielomianu: " << horner_it(stopien, wspolczynniki, x) << endl;
        
    
	
	return 0;
}

